src/site/xdoc/checks/misc/hexliteralcase.xml.template

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document xmlns="http://maven.apache.org/XDOC/2.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/XDOC/2.0 https://maven.apache.org/xsd/xdoc-2.0.xsd">
+  <head>
+    <title>HexLiteralCase</title>
+  </head>
+  <body>
+    <section name="HexLiteralCase">
+      <p>Since Checkstyle 12.1.0</p>
+
+      <subsection name="Description" id="HexLiteralCase_Description">
+        <macro name="description">
+          <param name="modulePath"
+                 value="src/main/java/com/puppycrawl/tools/checkstyle/checks/HexLiteralCaseCheck.java"/>
+        </macro>
+        <div>
+          Checks that hexadecimal literals are defined using uppercase letters <code>(A-F)</code>
```

Comment:
All description should be in java class, it should be a source for description.
Xdoc template should just use description macros to insert such data to website 

Please move all description text to javadoc of class